packages/menu-bar/src/vaadin-menu-bar-buttons-mixin.js

```diff
@@ -40,6 +50,18 @@ export const ButtonsMixin = (superClass) =>
       super.connectedCallback();
 
       this._initButtonAttrs(this._overflow);
+
+      // Observe parent node resize to detect overflow
+      this.__parent = this.parentNode instanceof ShadowRoot ? this.parentNode.host : this.parentNode;
+      this.__parentResizeObserver.observe(this.__parent);
+    }
+
+    /** @protected */
+    disconnectedCallback() {
+      super.disconnectedCallback();
+
+      this.__parentResizeObserver.unobserve(this.__parent);
```

Comment:
I'm not sure what would be a proper way to test this, maybe will have to use a long timeout 🤷‍♂️